working-groups/query-standardization/dsl-survey/ppl/README.md

```diff
@@ -0,0 +1,256 @@
+* **Diffing Support:**
+    * Comparisons over time can be done using appropriate queries and time filters, but specific "diffing" functionality is not inherently part of PPL.
+
+* * *
+**Events**
+
+* **Maximum Duration of Events Supported:**
+    * This depends on the data model and can potentially range from seconds to years.
+* **Event Model**:
+    *  Events must follow a structured data model, including start time and end time. Details can be similar to logs.
+
+**Traces (Distributed Tracing)**
+
+* **Tracing Models Supported:**
+    * Simple schema for Observability Supported both by Observability Plugin & Ingestion pipelines such as data-prepper & fluent-bit
+* **Maximum Duration of Traces Supported:**
+    *  Depending on the model, various durations are possible.
+* **Span Links:**
+    * If the data includes links between spans, these can be queried.
+    * [Observability-Dashboard](https://github.com/opensearch-project/dashboards-observability/wiki#trace-analytics) support these graphs out of the box.
+* **Limits:**
+    * May be constrained by the data model, indexing, and storage strategies.
+* **Querying Sub Graphs:**
+    *  Not direct API, possible through complex querying of the [span data](https://github.com/opensearch-project/opensearch-catalog/blob/main/schema/observability/traces/traces.mapping) index.
+* **Compound Predicates:**
+    * Supported based on how trace data is stored - mostly using the [span data](https://github.com/opensearch-project/opensearch-catalog/blob/main/schema/observability/traces/traces.mapping) index
+* **Joining Different Models:**
+    * Not directly supported - work in progress...
+* **Querying Incomplete Traces:**
+    * Partially supported - depending on data partiality
+* **Aggregations for Traces and Spans:**
+    *  Similar to logs; histograms, percentiles, max, min, sum, etc.
+    * Using OpenSearch [Transformation API](https://opensearch.org/docs/latest/im-plugin/index-transforms/index/) this can be materialized
+* **Attributes Typing at Query Time:**
+    *  Supported, with handling similar to logs.
+
+**Profiles (System Resources Measurement)**
+
+* **Supported Profiles and Models**:
+    * OpenSearch does not natively handle system profiles but can index and query profile data.
+    * Future support is being added using the OpenTelemetry semantic convention
+* **Filters:**
+    *  By duration, core, thread, function, environment variables, etc.
+* **Querying by DAG:**
+    * Partly Supported by leveraging the data-prepper ingestion time services graph enrichment.
+* **Creating "Diffs"**:
+    *  Can be partially achieved with appropriate queries - specific "diffing" functionality is not inherently part of PPL.
+* **Aggregating Multiple Profiles**:
+    *  Supported through appropriate query and aggregation techniques.
+
+**Metadata**
+
+* **Applying DSL for Telemetry to Metadata:**
+    * The DSL may be used to query metadata if it is stored and indexed in a compatible manner.
```

Comment:
Do you have an example or documentation? E.g. what would a Grafana plugin call when building a query?